pkg/statistics/handle/syncload/stats_syncload.go

```diff
@@ -122,25 +128,30 @@ func (*statsSyncLoad) SyncWaitStatsLoad(sc *stmtctx.StatementContext) error {
 	metrics.SyncLoadCounter.Inc()
 	timer := time.NewTimer(sc.StatsLoad.Timeout)
 	defer timer.Stop()
-	for {
+	for _, resultCh := range sc.StatsLoad.ResultCh {
 		select {
-		case result, ok := <-sc.StatsLoad.ResultCh:
+		case result, ok := <-resultCh:
 			if !ok {
 				return errors.New("sync load stats channel closed unexpectedly")
 			}
-			if result.HasError() {
-				errorMsgs = append(errorMsgs, result.ErrorMsg())
+			if result.Err != nil {
+				errorMsgs = append(errorMsgs, result.Err.Error())
 			}
-			delete(resultCheckMap, result.Item)
-			if len(resultCheckMap) == 0 {
-				metrics.SyncLoadHistogram.Observe(float64(time.Since(sc.StatsLoad.LoadStartTime).Milliseconds()))
-				return nil
+			val := result.Val.(stmtctx.StatsLoadResult)
+			if val.HasError() {
```

Comment:
Could you please add some comments to explain why we have one error from the result channel and one error from the value?

Reply:
Update.